pkgs/build-support/setup-hooks/patch-shebangs.sh

```diff
@@ -105,12 +105,12 @@ patchShebangs() {
 }
 
 patchShebangsAuto () {
-    if [ -z "${dontPatchShebangs-}" -a -e "$prefix" ]; then
+    if [[ -z "${dontPatchShebangs-}" && -e "$prefix" ]]; then
```

Comment:
I've noticed `dontPatchShebangs` [not being respected](https://github.com/NixOS/nixpkgs/issues/117985) for the `androidStudioPackages` a bit ago, could this have anything to do with this change? 

I'm not too familair with the stdenv or bash scripting, so please disregard this comment if it does not make any sense.
```suggestion
    if [[ -z "${dontPatchShebangs-}" || -e "$prefix" ]]; then
```

Reply:
The `-a` stands for `&&`. Is `dontPatchShebangs` unset by anything in the package you mention?